x-pack/legacy/plugins/graph/public/components/graph_settings/advanced_settings_form.tsx

```diff
@@ -0,0 +1,214 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React from 'react';
+import { EuiFormRow, EuiFieldNumber, EuiComboBox, EuiSwitch, EuiText } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { GraphSettingsProps } from './graph_settings';
+import { AdvancedSettings } from '../../types';
+
+// Helper type to get all keys of an interface
+// that are of type number.
+// Used to parameterize the number updater function
+// to make it impossible to pass in non-number keys
+type NumberKeys<T> = Exclude<
+  {
+    [K in keyof T]: T[K] extends number ? K : never;
+  }[keyof T],
+  undefined
+>;
+
+export function AdvancedSettingsForm({
+  advancedSettings,
+  updateAdvancedSettings,
+  allFields,
+}: Pick<GraphSettingsProps, 'advancedSettings' | 'updateAdvancedSettings' | 'allFields'>) {
+  function updateSetting<K extends keyof AdvancedSettings>(key: K, value: AdvancedSettings[K]) {
+    updateAdvancedSettings({ ...advancedSettings, [key]: value });
+  }
+
+  function getNumberUpdater<K extends NumberKeys<AdvancedSettings>>(key: K) {
+    return function({ target: { valueAsNumber } }: { target: { valueAsNumber: number } }) {
+      updateSetting(key, Number.isNaN(valueAsNumber) ? 0 : valueAsNumber);
+    };
+  }
+
+  return (
+    <form>
+      <EuiFormRow
+        fullWidth
+        helpText={i18n.translate('xpack.graph.settings.advancedSettings.sampleSizeInputHelpText', {
+          defaultMessage:
+            'Terms are identified from samples of the most relevant documents. Bigger is not necessarily better - can be slower and less relevant',
+        })}
+        label={i18n.translate('xpack.graph.settings.advancedSettings.sampleSizeInputLabel', {
+          defaultMessage: 'Sample size',
+        })}
+      >
+        <EuiFieldNumber
```

Comment:
**Optional Enhancement**: Use an EuiRange with `showInput=true`

Since there is a min and max value set for this control, you could leverage the UI to supply a slider.